### DISCUSSION TITLE
Help in updating example since keras-nlp API change

### DISCUSSION THREAD

**Discussion by jefke24** [keras-team/keras-hub#1035] [Q&A] [answered]
Hi All,

I'm a beginner trying to follow along some examples from Keras-nlp, more specifically this one: 
https://github.com/keras-team/keras-io/blob/master/examples/generative/text_generation_gpt.py

However, since the last update and API change, this doesn't work anymore.
More specifically, since keras_nlp.utils.top_p_search has been replaced with the TopPSampler class.

Can anyone clarify how I would get that example to work using TopPSampler?

Now, top-p-search is based on this:

`# Unpadded bos token.
prompt_tokens = tf.convert_to_tensor([tokenizer.token_to_id("[BOS]")])

def token_logits_fn(inputs):
    cur_len = inputs.shape[1]
    output = model(inputs)
    return output[:, cur_len - 1, :]  # return next token logits

output_tokens = keras_nlp.utils.top_p_search(
    token_logits_fn,
    prompt_tokens,
    max_length=NUM_TOKENS_TO_GENERATE,
    p=0.5,
    from_logits=True,
)
txt = tokenizer.detokenize(output_tokens)
print(f"Top-P search generated text: \n{txt}\n")`

I'm struggling to get this reworked to the new API.

**Comment by jbischof** [comment, marked as answer]
See #1030. We will update the old guide ASAP, apologies!

**Reply by jefke24** [reply]
Djeez, no need to apologize or I'm going to start feeling guilty :-)
Thanks a lot for the feedback, looking forward to the new version!
